core/container/src/main/java/org/wildfly/swarm/internal/FileSystemLayout.java

```diff
@@ -76,6 +79,40 @@ public static FileSystemLayout create() {
      */
     public static FileSystemLayout create(String root) {
 
+        // THORN-2178: Check if a System property has been set to a specific implementation class.
+        String implClassName = System.getProperty(FileSystemLayout.class.getName());
```

Comment:
Thought about it ... then realized that the class itself was internal. This scenario would be very rare and we should be able to use the class name itself. So if the packages get renamed in the future, this would automatically change. WDYT?

Reply:
@manovotn The PR needs a resolution on this issue alone now. What do you suggest the default behavior ought to be?

Reply:
> So if the packages get renamed in the future, this would automatically change

That's exactly why I wouldn't bind to to the class itself. You usually want to have these properties documented somewhere and don't want them to change as a side-effect of internal code changes.

So I would probably make a public constant with `thorntail` in its name.